shared/ipldutils.go

```diff
@@ -0,0 +1,289 @@
+package shared
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/codec/dagcbor"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/ipld/go-ipld-prime/schema"
+	cbg "github.com/whyrusleeping/cbor-gen"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/crypto"
+)
+
+type typeWithBindnodeSchema interface {
+	BindnodeSchema() string
```

Comment:
yep, that's the next step, refactor this stuff out of here and figure out the minimum necessary useful set of API

I was thinking ipld/go-ipld-prime/node/bindnode/util
or even just a ipld/go-bindnode-utils package, because we could even collect some common type converters in there too perhaps